crates/polars-core/src/series/arithmetic/borrowed.rs

```diff
@@ -171,6 +171,50 @@ impl NumOpsDispatchInner for FixedSizeListType {
     }
 }
 
+impl ListChunked {
+    fn arithm_helper(
+        &self,
+        rhs: &Series,
+        op: &dyn Fn(&Series, &Series) -> PolarsResult<Series>,
+    ) -> PolarsResult<Series> {
+        polars_ensure!(self.len() == rhs.len(), InvalidOperation: "can only do arithmetic operations on Series of the same size; got {} and {}", self.len(), rhs.len());
+
+        let mut result = self.clear();
+        let combined = self.amortized_iter().zip(rhs.list()?.amortized_iter()).map(|(a, b)| {
```

Comment:
I really think we should work on leaves and later add the `offset`s. I want to multiply the base arrays without the nesting and later restore the proper nesting structure.

Reply:
I did that by adding the leaf values array directly, unchanged, and ... then I realized that as soon as you have nulls adding those two together is no longer necessarily correct:

```python
>>> df = pl.Series([[1, 2], None, [3, 4]])
>>> df2 = pl.Series([None, [1, 2], [3, 4]])
>>> df + df2
shape: (3,)
Series: '' [list[i64]]
[
        [2, 4]
        null
        [6, 8]
]
```

In the underlying Arrow representation there's an (Arrow) values array of length 4 for each... but the offsets are different when the nulls don't line up.

That result is wrong. So if there are any nulls you fall back to something using `amortized_iter()`. I'm fairly certain the code as originally submitted also handles nulls incorrectly, but in a different way... so first I'm going to write a bunch of tests.

Currently thinking there will be two code paths, a fast path for no nulls (adds the underlying values arrays), and a slow path for when there are nulls (the logic in the original PR, more or less).